<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>References | Network Medicine</title>
  <meta name="description" content="Workshop in Advanced Bioinformatics: Network Medicine." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="References | Network Medicine" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Workshop in Advanced Bioinformatics: Network Medicine." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | Network Medicine" />
  
  <meta name="twitter:description" content="Workshop in Advanced Bioinformatics: Network Medicine." />
  

<meta name="author" content="Deisy Morselli Gysi" />


<meta name="date" content="2021-02-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="summary.html"/>

<script src="libs/header-attrs-2.6.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Network Medicine</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="workshop-on-network-medicine.html"><a href="workshop-on-network-medicine.html"><i class="fa fa-check"></i><b>1</b> Workshop on Network Medicine</a>
<ul>
<li class="chapter" data-level="1.1" data-path="workshop-on-network-medicine.html"><a href="workshop-on-network-medicine.html#whatarenets"><i class="fa fa-check"></i><b>1.1</b> What are networks?</a></li>
<li class="chapter" data-level="1.2" data-path="workshop-on-network-medicine.html"><a href="workshop-on-network-medicine.html#terminology"><i class="fa fa-check"></i><b>1.2</b> Terminology</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="workshop-on-network-medicine.html"><a href="workshop-on-network-medicine.html#network-terminology"><i class="fa fa-check"></i><b>1.2.1</b> Network Terminology</a></li>
<li class="chapter" data-level="1.2.2" data-path="workshop-on-network-medicine.html"><a href="workshop-on-network-medicine.html#biological-terminology"><i class="fa fa-check"></i><b>1.2.2</b> Biological Terminology</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html"><i class="fa fa-check"></i><b>2</b> Data Commonly Used in Network Medicine</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#PPI"><i class="fa fa-check"></i><b>2.1</b> Protein Protein Interaction Networks</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#measuring-ppis"><i class="fa fa-check"></i><b>2.1.1</b> Measuring PPIs</a></li>
<li class="chapter" data-level="2.1.2" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#commonly-used-data-sources-for-ppis"><i class="fa fa-check"></i><b>2.1.2</b> Commonly used data sources for PPIs</a></li>
<li class="chapter" data-level="2.1.3" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#understanding-a-ppi"><i class="fa fa-check"></i><b>2.1.3</b> Understanding a PPI</a></li>
<li class="chapter" data-level="2.1.4" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#exercises"><i class="fa fa-check"></i><b>2.1.4</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#GDA"><i class="fa fa-check"></i><b>2.2</b> Gene Disease Association</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#commonly-used-data-sources-for-gdas"><i class="fa fa-check"></i><b>2.2.1</b> Commonly used data sources for GDAs</a></li>
<li class="chapter" data-level="2.2.2" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#understanding-a-gda-dataset"><i class="fa fa-check"></i><b>2.2.2</b> Understanding a GDA dataset</a></li>
<li class="chapter" data-level="2.2.3" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#exercises-1"><i class="fa fa-check"></i><b>2.2.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#DTs"><i class="fa fa-check"></i><b>2.3</b> Drug-Targets</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#properties-of-an-ideal-drug-target"><i class="fa fa-check"></i><b>2.3.1</b> Properties of an ideal drug target:</a></li>
<li class="chapter" data-level="2.3.2" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#commonly-used-data-sources-for-gdas-1"><i class="fa fa-check"></i><b>2.3.2</b> Commonly used data sources for GDAs</a></li>
<li class="chapter" data-level="2.3.3" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#understanding-a-drug-target-dataset"><i class="fa fa-check"></i><b>2.3.3</b> Understanding a Drug-Target dataset</a></li>
<li class="chapter" data-level="2.3.4" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#exercises-2"><i class="fa fa-check"></i><b>2.3.4</b> Exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>3</b> Methods fo Disease Module Identification and Disease Similarity</a>
<ul>
<li class="chapter" data-level="3.1" data-path="methods.html"><a href="methods.html#diseasemodule"><i class="fa fa-check"></i><b>3.1</b> Disease Module</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="methods.html"><a href="methods.html#largest-connected-component"><i class="fa fa-check"></i><b>3.1.1</b> Largest connected component</a></li>
<li class="chapter" data-level="3.1.2" data-path="methods.html"><a href="methods.html#example-in-real-data"><i class="fa fa-check"></i><b>3.1.2</b> Example in real data</a></li>
<li class="chapter" data-level="3.1.3" data-path="methods.html"><a href="methods.html#exercises-3"><i class="fa fa-check"></i><b>3.1.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="methods.html"><a href="methods.html#gene-overlap"><i class="fa fa-check"></i><b>3.2</b> Gene Overlap</a></li>
<li class="chapter" data-level="3.3" data-path="methods.html"><a href="methods.html#dissep"><i class="fa fa-check"></i><b>3.3</b> Disease Separation</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="methods.html"><a href="methods.html#example-in-real-data-1"><i class="fa fa-check"></i><b>3.3.1</b> Example in real data</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="methods.html"><a href="methods.html#exercises-4"><i class="fa fa-check"></i><b>3.4</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="method-for-drug-repurpusing.html"><a href="method-for-drug-repurpusing.html"><i class="fa fa-check"></i><b>4</b> Method for drug-repurpusing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="method-for-drug-repurpusing.html"><a href="method-for-drug-repurpusing.html#proximity"><i class="fa fa-check"></i><b>4.1</b> Proximity</a></li>
<li class="chapter" data-level="4.2" data-path="method-for-drug-repurpusing.html"><a href="method-for-drug-repurpusing.html#example-in-real-data-2"><i class="fa fa-check"></i><b>4.2</b> Example in real data</a></li>
<li class="chapter" data-level="4.3" data-path="method-for-drug-repurpusing.html"><a href="method-for-drug-repurpusing.html#exercises-5"><i class="fa fa-check"></i><b>4.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="summary.html"><a href="summary.html"><i class="fa fa-check"></i><b>5</b> Summary</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="./">Workshop in Advanced Bioinformatics</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Network Medicine</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered">
<h1>References</h1>

<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Ahn2010" class="csl-entry">
Ahn, Yong Yeol, James P. Bagrow, and Sune Lehmann. 2010. <span>“<span class="nocase">Link communities reveal multiscale complexity in networks</span>.”</span> <em>Nature</em>. <a href="https://doi.org/10.1038/nature09182">https://doi.org/10.1038/nature09182</a>.
</div>
<div id="ref-Alanis-Lobato2017" class="csl-entry">
Alanis-Lobato, Gregorio, Miguel A Andrade-Navarro, and Martin H Schaefer. 2017. <span>“<span class="nocase">HIPPIE v2.0: enhancing meaningfulness and reliability of protein-protein interaction networks.</span>”</span> <em>Nucleic Acids Research</em> 45 (D1): D408–14. <a href="https://doi.org/10.1093/nar/gkw985">https://doi.org/10.1093/nar/gkw985</a>.
</div>
<div id="ref-Alonso-Lopez2019a" class="csl-entry">
Alonso-López, DIego, Francisco J. Campos-Laborie, Miguel A. Gutiérrez, Luke Lambourne, Michael A. Calderwood, Marc Vidal, and Javier De Las Rivas. 2019. <span>“<span class="nocase">APID database: Redefining protein-protein interaction experimental evidences and binary interactomes</span>.”</span> <em>Database</em> 2019 (i): 1–8. <a href="https://doi.org/10.1093/database/baz005">https://doi.org/10.1093/database/baz005</a>.
</div>
<div id="ref-barabasi1999emergence" class="csl-entry">
Barabási, Albert László, and Réka Albert. 1999. <span>“<span class="nocase">Emergence of scaling in random networks</span>.”</span> <em>Science</em> 286 (5439): 509–12. <a href="https://doi.org/10.1126/science.286.5439.509">https://doi.org/10.1126/science.286.5439.509</a>.
</div>
<div id="ref-barabasi2004network" class="csl-entry">
Barabási, Albert László, and Zoltán N. Oltvai. 2004. <span>“<span class="nocase">Network biology: Understanding the cell’s functional organization</span>.”</span> <em>Nature Reviews Genetics</em> 5 (2): 101–13. <a href="https://doi.org/10.1038/nrg1272">https://doi.org/10.1038/nrg1272</a>.
</div>
<div id="ref-Barabasi2016" class="csl-entry">
Barabási, Albert-Láslzo. 2016. <em><span>Network Science</span></em>. 1st ed. Cambridge, UK: Cambridge University Press.
</div>
<div id="ref-Barrat2003TheNetworks" class="csl-entry">
Barrat, Alain, Marc Barthelemy, Romualdo Pastor-Satorras, and Alessandro Vespignani. 2003. <span>“<span class="nocase">The architecture of complex weighted networks</span>.”</span> <em>Proceedings of the National Academy of Sciences</em> 101 (11): 3747–52. <a href="https://doi.org/10.1073/pnas.0400087101">https://doi.org/10.1073/pnas.0400087101</a>.
</div>
<div id="ref-Bondy2008GraphTheory" class="csl-entry">
Bondy, J. A., and U. S. R. Murty. 2008. <em><span>Graph Theory</span></em>. Springer. <a href="https://doi.org/10.1007/978-1-84638-970-5">https://doi.org/10.1007/978-1-84638-970-5</a>.
</div>
<div id="ref-Borgatti2006ACentrality" class="csl-entry">
Borgatti, Stephen P., and Martin G. Everett. 2006. <span>“<span class="nocase">A Graph-theoretic perspective on centrality</span>.”</span> <em>Social Networks</em> 28 (4): 466–84. <a href="https://doi.org/10.1016/J.SOCNET.2005.11.005">https://doi.org/10.1016/J.SOCNET.2005.11.005</a>.
</div>
<div id="ref-Breuer2013" class="csl-entry">
Breuer, Karin, Amir K Foroushani, Matthew R Laird, Carol Chen, Anastasia Sribnaia, Raymond Lo, Geoffrey L Winsor, Robert E W Hancock, Fiona S L Brinkman, and David J Lynn. 2013. <span>“<span class="nocase">InnateDB: systems biology of innate immunity and beyond–recent updates and continuing curation.</span>”</span> <em>Nucleic Acids Research</em> 41 (Database issue): D1228–33. <a href="https://doi.org/10.1093/nar/gks1147">https://doi.org/10.1093/nar/gks1147</a>.
</div>
<div id="ref-buniello2018" class="csl-entry">
Buniello, Annalisa, Jacqueline A L MacArthur, Maria Cerezo, Laura W Harris, James Hayhurst, Cinzia Malangone, Aoife McMahon, et al. 2018. <span>“The NHGRI-EBI GWAS Catalog of Published Genome-Wide Association Studies, Targeted Arrays and Summary Statistics 2019.”</span> <em>Nucleic Acids Research</em> 47 (D1): D1005–12. <a href="https://doi.org/10.1093/nar/gky1120">https://doi.org/10.1093/nar/gky1120</a>.
</div>
<div id="ref-Cao2013" class="csl-entry">
Cao, Mengfei, Hao Zhang, Jisoo Park, Noah M. Daniels, Mark E. Crovella, Lenore J. Cowen, and Benjamin Hescott. 2013. <span>“<span class="nocase">Going the Distance for Protein Function Prediction: A New Distance Metric for Protein Interaction Networks</span>.”</span> <em>PLoS ONE</em>. <a href="https://doi.org/10.1371/journal.pone.0076339">https://doi.org/10.1371/journal.pone.0076339</a>.
</div>
<div id="ref-Chatr-Aryamontri2017" class="csl-entry">
Chatr-Aryamontri, Andrew, Rose Oughtred, Lorrie Boucher, Jennifer Rust, Christie Chang, Nadine K Kolas, Lara O’Donnell, et al. 2017. <span>“<span class="nocase">The BioGRID interaction database: 2017 update.</span>”</span> <em>Nucleic Acids Research</em> 45 (D1): D369–79. <a href="https://doi.org/10.1093/nar/gkw1102">https://doi.org/10.1093/nar/gkw1102</a>.
</div>
<div id="ref-chen2012" class="csl-entry">
Chen, Geng, Ziyun Wang, Dongqing Wang, Chengxiang Qiu, Mingxi Liu, Xing Chen, Qipeng Zhang, Guiying Yan, and Qinghua Cui. 2012. <span>“LncRNADisease: A Database for Long-Non-Coding RNA-Associated Diseases.”</span> <em>Nucleic Acids Research</em> 41 (D1): D983–86. <a href="https://doi.org/10.1093/nar/gks1099">https://doi.org/10.1093/nar/gks1099</a>.
</div>
<div id="ref-cheng2014" class="csl-entry">
Cheng, Feixiong, Peilin Jia, Quan Wang, and Zhongming Zhao. 2014. <span>“Quantitative Network Mapping of the Human Kinome Interactome Reveals New Clues for Rational Kinase Inhibitor Discovery and Individualized Cancer Therapy.”</span> <em>Oncotarget</em> 5 (11): 3697–3710. <a href="https://doi.org/10.18632/oncotarget.1984">https://doi.org/10.18632/oncotarget.1984</a>.
</div>
<div id="ref-corsello2017" class="csl-entry">
Corsello, Steven M, Joshua A Bittker, Zihan Liu, Joshua Gould, Patrick McCarren, Jodi E Hirschman, Stephen E Johnston, et al. 2017. <span>“The Drug Repurposing Hub: A Next-Generation Drug Library and Information Resource.”</span> <em>Nature Medicine</em> 23 (4): 405–8. <a href="https://doi.org/10.1038/nm.4306">https://doi.org/10.1038/nm.4306</a>.
</div>
<div id="ref-Cowley2012" class="csl-entry">
Cowley, Mark J, Mark Pinese, Karin S Kassahn, Nic Waddell, John V Pearson, Sean M Grimmond, Andrew V Biankin, Sampsa Hautaniemi, and Jianmin Wu. 2012. <span>“<span class="nocase">PINA v2.0: mining interactome modules.</span>”</span> <em>Nucleic Acids Research</em> 40 (Database issue): D862–5. <a href="https://doi.org/10.1093/nar/gkr967">https://doi.org/10.1093/nar/gkr967</a>.
</div>
<div id="ref-Das2012" class="csl-entry">
Das, Jishnu, and Haiyuan Yu. 2012. <span>“<span class="nocase">HINT: High-quality protein interactomes and their applications in understanding human disease</span>.”</span> <em>BMC Systems Biology</em> 6. <a href="https://doi.org/10.1186/1752-0509-6-92">https://doi.org/10.1186/1752-0509-6-92</a>.
</div>
<div id="ref-davis2020" class="csl-entry">
Davis, Allan Peter, Cynthia J Grondin, Robin J Johnson, Daniela Sciaky, Jolene Wiegers, Thomas C Wiegers, and Carolyn J Mattingly. 2020. <span>“Comparative Toxicogenomics Database (CTD): Update 2021.”</span> <em>Nucleic Acids Research</em> 49 (D1): D1138–43. <a href="https://doi.org/10.1093/nar/gkaa891">https://doi.org/10.1093/nar/gkaa891</a>.
</div>
<div id="ref-DeDomenico2017" class="csl-entry">
De Domenico, Manlio. 2017. <span>“<span class="nocase">Multilayer modeling and analysis of human brain networks</span>.”</span> <em>GigaScience</em> 6 (5). <a href="https://doi.org/10.1093/gigascience/gix004">https://doi.org/10.1093/gigascience/gix004</a>.
</div>
<div id="ref-Fazekas2013" class="csl-entry">
Fazekas, Dávid, Mihály Koltai, Dénes Türei, Dezső Módos, Máté Pálfy, Zoltán Dúl, Lilian Zsákai, et al. 2013. <span>“<span class="nocase">SignaLink 2 - a signaling pathway resource with multi-layered regulatory networks.</span>”</span> <em>BMC Systems Biology</em> 7 (January): 7. <a href="https://doi.org/10.1186/1752-0509-7-7">https://doi.org/10.1186/1752-0509-7-7</a>.
</div>
<div id="ref-fong2004" class="csl-entry">
Fong, Jessica H, Amy E Keating, and Mona Singh. 2004. <em>Genome Biology</em> 5 (2): R11. <a href="https://doi.org/10.1186/gb-2004-5-2-r11">https://doi.org/10.1186/gb-2004-5-2-r11</a>.
</div>
<div id="ref-Gashaw2011" class="csl-entry">
Gashaw, Isabella, Peter Ellinghaus, Anette Sommer, and Khusru Asadullah. 2011. <span>“<span class="nocase">What makes a good drug target?</span>”</span> <a href="https://doi.org/10.1016/j.drudis.2011.09.007">https://doi.org/10.1016/j.drudis.2011.09.007</a>.
</div>
<div id="ref-Guney2016" class="csl-entry">
Guney, Emre, Jörg Menche, Marc Vidal, and Albert-László László Barábasi. 2016. <span>“<span class="nocase">Network-based in silico drug efficacy screening</span>.”</span> <em>Nature Communications</em> 7 (1): 10331. <a href="https://doi.org/10.1038/ncomms10331">https://doi.org/10.1038/ncomms10331</a>.
</div>
<div id="ref-Gysi2020a" class="csl-entry">
Gysi, D M, Í Do Valle Zitnik, M., A Ameli, X Gan, O Varol, H Sanchez, R M Baron, D Ghiassian, J Loscalzo, and A L Barabási. 2020. <span>“<span class="nocase">Network Medicine Framework for Identifying Drug Repurposing Opportunities for COVID-19</span>.”</span> <em>ArXiv</em>.
</div>
<div id="ref-Gysi2020" class="csl-entry">
Gysi, Deisy Morselli, and Katja Nowick. 2020. <span>“<span class="nocase">Construction, comparison and evolution of networks in life sciences and other disciplines</span>.”</span> <em>Journal of the Royal Society, Interface</em> 17 (166): 20190610. <a href="https://doi.org/10.1098/rsif.2019.0610">https://doi.org/10.1098/rsif.2019.0610</a>.
</div>
<div id="ref-Hein2015" class="csl-entry">
Hein, Marco Y., Nina C. Hubner, Ina Poser, Jürgen Cox, Nagarjuna Nagaraj, Yusuke Toyoda, Igor A. Gak, et al. 2015. <span>“<span class="nocase">A Human Interactome in Three Quantitative Dimensions Organized by Stoichiometries and Abundances</span>.”</span> <em>Cell</em> 163 (3): 712–23. <a href="https://doi.org/10.1016/j.cell.2015.09.053">https://doi.org/10.1016/j.cell.2015.09.053</a>.
</div>
<div id="ref-Hobert2008GeneMicroRNAs" class="csl-entry">
Hobert, Oliver. 2008. <span>“<span class="nocase">Gene regulation by transcription factors and MicroRNAs</span>.”</span> American Association for the Advancement of Science. <a href="https://doi.org/10.1126/science.1151651">https://doi.org/10.1126/science.1151651</a>.
</div>
<div id="ref-Hornbeck2015" class="csl-entry">
Hornbeck, Peter V, Bin Zhang, Beth Murray, Jon M Kornhauser, Vaughan Latham, and Elzbieta Skrzypek. 2015. <span>“<span class="nocase">PhosphoSitePlus, 2014: mutations, PTMs and recalibrations.</span>”</span> <em>Nucleic Acids Research</em> 43 (Database issue): D512–20. <a href="https://doi.org/10.1093/nar/gku1267">https://doi.org/10.1093/nar/gku1267</a>.
</div>
<div id="ref-huang2018" class="csl-entry">
Huang, Zhou, Jiangcheng Shi, Yuanxu Gao, Chunmei Cui, Shan Zhang, Jianwei Li, Yuan Zhou, and Qinghua Cui. 2018. <span>“HMDD V3.0: A Database for Experimentally Supported Human microRNA<span><span></span></span>disease Associations.”</span> <em>Nucleic Acids Research</em> 47 (D1): D1013–17. <a href="https://doi.org/10.1093/nar/gky1010">https://doi.org/10.1093/nar/gky1010</a>.
</div>
<div id="ref-Huttlin2017" class="csl-entry">
Huttlin, Edward L., Raphael J. Bruckner, Joao A. Paulo, Joe R. Cannon, Lily Ting, Kurt Baltier, Greg Colby, et al. 2017. <span>“<span class="nocase">Architecture of the human interactome defines protein communities and disease networks</span>.”</span> <em>Nature</em> 545 (7655): 505–9. <a href="https://doi.org/10.1038/nature22366">https://doi.org/10.1038/nature22366</a>.
</div>
<div id="ref-Kempa2019HighAnalysis" class="csl-entry">
Kempa, Emily E., Katherine A. Hollywood, Clive A. Smith, and Perdita E. Barran. 2019. <span>“<span class="nocase">High throughput screening of complex biological samples with mass spectrometry-from bulk measurements to single cell analysis</span>.”</span> <em>Analyst</em> 144 (3): 872–91. <a href="https://doi.org/10.1039/c8an01448e">https://doi.org/10.1039/c8an01448e</a>.
</div>
<div id="ref-Kivela2014MultilayerNetworks" class="csl-entry">
Kivela, M., A. Arenas, M. Barthelemy, J. P. Gleeson, Y. Moreno, and M. A. Porter. 2014. <span>“<span class="nocase">Multilayer networks</span>.”</span> <em>Journal of Complex Networks</em> 2 (3): 203–71. <a href="https://doi.org/10.1093/comnet/cnu016">https://doi.org/10.1093/comnet/cnu016</a>.
</div>
<div id="ref-koh2012" class="csl-entry">
Koh, Gavin C. K. W., Pablo Porras, Bruno Aranda, Henning Hermjakob, and Sandra E. Orchard. 2012. <span>“Analyzing Protein<span><span></span></span>Protein Interaction Networks.”</span> <em>Journal of Proteome Research</em> 11 (4): 2014–31. <a href="https://doi.org/10.1021/pr201211w">https://doi.org/10.1021/pr201211w</a>.
</div>
<div id="ref-Kurant2006LayeredNetworks" class="csl-entry">
Kurant, Maciej, and Patrick Thiran. 2006. <span>“<span>Layered Complex Networks</span>.”</span> <em>Physical Review Letters</em> 96 (13): 138701. <a href="https://doi.org/10.1103/PhysRevLett.96.138701">https://doi.org/10.1103/PhysRevLett.96.138701</a>.
</div>
<div id="ref-landrum2019" class="csl-entry">
Landrum, Melissa J, Shanmuga Chitipiralla, Garth R Brown, Chao Chen, Baoshan Gu, Jennifer Hart, Douglas Hoffman, et al. 2019. <span>“ClinVar: Improvements to Accessing Data.”</span> <em>Nucleic Acids Research</em> 48 (D1): D835–44. <a href="https://doi.org/10.1093/nar/gkz972">https://doi.org/10.1093/nar/gkz972</a>.
</div>
<div id="ref-Latchman1997TranscriptionOverview" class="csl-entry">
Latchman, D. S. 1997. <span>“<span class="nocase">Transcription factors: An overview</span>.”</span> <em>International Journal of Biochemistry and Cell Biology</em> 29 (12): 1305–12. <a href="https://doi.org/10.1016/S1357-2725(97)00085-X">https://doi.org/10.1016/S1357-2725(97)00085-X</a>.
</div>
<div id="ref-Li2009" class="csl-entry">
Li, Ai, and Steve Horvath. 2009. <span>“<span class="nocase">Network module detection: Affinity search technique with the multi-node topological overlap measure</span>.”</span> <em>BMC Research Notes</em> 2 (1): 142. <a href="https://doi.org/10.1186/1756-0500-2-142">https://doi.org/10.1186/1756-0500-2-142</a>.
</div>
<div id="ref-li2016" class="csl-entry">
Li, Taibo, Rasmus Wernersson, Rasmus B Hansen, Heiko Horn, Johnathan Mercer, Greg Slodkowicz, Christopher T Workman, et al. 2016. <span>“A Scored Human Protein<span><span></span></span>protein Interaction Network to Catalyze Genomic Interpretation.”</span> <em>Nature Methods</em> 14 (1): 61–64. <a href="https://doi.org/10.1038/nmeth.4083">https://doi.org/10.1038/nmeth.4083</a>.
</div>
<div id="ref-Licata2012" class="csl-entry">
Licata, Luana, Leonardo Briganti, Daniele Peluso, Livia Perfetto, Marta Iannuccelli, Eugenia Galeota, Francesca Sacco, et al. 2012. <span>“<span class="nocase">MINT, the molecular interaction database: 2012 update</span>.”</span> <em>Nucleic Acids Research</em> 40 (D1): D857–61. <a href="https://doi.org/10.1093/nar/gkr930">https://doi.org/10.1093/nar/gkr930</a>.
</div>
<div id="ref-Luck2020" class="csl-entry">
Luck, Katja, Dae Kyum Kim, Luke Lambourne, Kerstin Spirohn, Bridget E. Begg, Wenting Bian, Ruth Brignall, et al. 2020. <span>“<span class="nocase">A reference map of the human binary protein interactome</span>.”</span> <em>Nature</em>. <a href="https://doi.org/10.1038/s41586-020-2188-x">https://doi.org/10.1038/s41586-020-2188-x</a>.
</div>
<div id="ref-MattickNon-codingRNA" class="csl-entry">
Mattick, John S, and Igor V Makunin. 2006. <span>“<span class="nocase">Non-coding RNA</span>.”</span> <em>Human Molecular Genetics</em> 15 (suppl<span>_</span>1): R17–29. <a href="https://doi.org/10.1093/hmg/ddl046">https://doi.org/10.1093/hmg/ddl046</a>.
</div>
<div id="ref-mckusick2007" class="csl-entry">
McKusick, Victor A. 2007. <span>“Mendelian Inheritance in Man and Its Online Version, OMIM.”</span> <em>The American Journal of Human Genetics</em> 80 (4): 588–604. <a href="https://doi.org/10.1086/514346">https://doi.org/10.1086/514346</a>.
</div>
<div id="ref-Menche2015" class="csl-entry">
Menche, Jörg, Amitabh Sharma, Maksim Kitsak, Susan Dina Ghiassian, Marc Vidal, Joseph Loscalzo, and Albert-Laszlo Barabasi. 2015. <span>“<span class="nocase">Uncovering disease-disease relationships through the incomplete interactome</span>.”</span> <em>Science</em> 347 (6224). <a href="https://doi.org/10.1126/science.1065103">https://doi.org/10.1126/science.1065103</a>.
</div>
<div id="ref-Metzker2010SequencingGeneration" class="csl-entry">
Metzker, Michael L. 2010. <span>“<span class="nocase">Sequencing technologies the next generation</span>.”</span> Nature Publishing Group. <a href="https://doi.org/10.1038/nrg2626">https://doi.org/10.1038/nrg2626</a>.
</div>
<div id="ref-Meyer2018" class="csl-entry">
Meyer, Michael J, Juan Felipe Beltrán, Siqi Liang, Robert Fragoza, Aaron Rumack, Jin Liang, Xiaomu Wei, and Haiyuan Yu. 2018. <span>“<span class="nocase">Interactome INSIDER: a structural interactome browser for genomic studies.</span>”</span> <em>Nature Methods</em> 15 (2): 107–14. <a href="https://doi.org/10.1038/nmeth.4540">https://doi.org/10.1038/nmeth.4540</a>.
</div>
<div id="ref-Meyer2013" class="csl-entry">
Meyer, Michael J, Jishnu Das, Xiujuan Wang, and Haiyuan Yu. 2013. <span>“<span class="nocase">INstruct: a database of high-quality 3D structurally resolved protein interactome networks.</span>”</span> <em>Bioinformatics (Oxford, England)</em> 29 (12): 1577–79. <a href="https://doi.org/10.1093/bioinformatics/btt181">https://doi.org/10.1093/bioinformatics/btt181</a>.
</div>
<div id="ref-Mosca2013" class="csl-entry">
Mosca, Roberto, Arnaud Céol, and Patrick Aloy. 2013. <span>“<span class="nocase">Interactome3D: adding structural details to protein networks.</span>”</span> <em>Nature Methods</em> 10 (1): 47–53. <a href="https://doi.org/10.1038/nmeth.2289">https://doi.org/10.1038/nmeth.2289</a>.
</div>
<div id="ref-Mucha2010CommunityNetworks" class="csl-entry">
Mucha, Peter J, Thomas Richardson, Kevin Macon, Mason A Porter, and Jukka-Pekka Onnela. 2010. <span>“<span class="nocase">Community structure in time-dependent, multiscale, and multiplex networks.</span>”</span> <em>Science (New York, N.Y.)</em> 328 (5980): 876–78. <a href="https://doi.org/10.1126/science.1184819">https://doi.org/10.1126/science.1184819</a>.
</div>
<div id="ref-Newman2018" class="csl-entry">
Newman, Mark E J. 2018. <em><span>Networks</span></em>. <a href="https://books.google.com.br/books?hl=en&amp;lr=&amp;id=YdZjDwAAQBAJ&amp;oi=fnd&amp;pg=PP1&amp;dq=networks+an+introduction&amp;ots=V_J-6Ig5uu&amp;sig=wlDx19lsqamr-20U5ZUbEjwW3uM#v=onepage&amp;q=networks an introduction&amp;f=false https://www.ebooks.com/96308191/networks/newman-mark/">https://books.google.com.br/books?hl=en&amp;lr=&amp;id=YdZjDwAAQBAJ&amp;oi=fnd&amp;pg=PP1&amp;dq=networks+an+introduction&amp;ots=V_J-6Ig5uu&amp;sig=wlDx19lsqamr-20U5ZUbEjwW3uM#v=onepage&amp;q=networks an introduction&amp;f=false https://www.ebooks.com/96308191/networks/newman-mark/</a>.
</div>
<div id="ref-Park2009ChIP-seq:Technology" class="csl-entry">
Park, Peter J. 2009. <span>“<span class="nocase">ChIP-seq: Advantages and challenges of a maturing technology</span>.”</span> Nature Publishing Group. <a href="https://doi.org/10.1038/nrg2641">https://doi.org/10.1038/nrg2641</a>.
</div>
<div id="ref-Parrish2006YeastMapping" class="csl-entry">
Parrish, Jodi R., Keith D. Gulyas, and Russell L. Finley. 2006. <span>“<span class="nocase">Yeast two-hybrid contributions to interactome mapping</span>.”</span> Elsevier Current Trends. <a href="https://doi.org/10.1016/j.copbio.2006.06.006">https://doi.org/10.1016/j.copbio.2006.06.006</a>.
</div>
<div id="ref-piñero2019" class="csl-entry">
Piñero, Janet, Juan Manuel Ramírez-Anguita, Josep Saüch-Pitarch, Francesco Ronzano, Emilio Centeno, Ferran Sanz, and Laura I Furlong. 2019. <span>“The DisGeNET Knowledge Platform for Disease Genomics: 2019 Update.”</span> <em>Nucleic Acids Research</em>, November. <a href="https://doi.org/10.1093/nar/gkz1021">https://doi.org/10.1093/nar/gkz1021</a>.
</div>
<div id="ref-Radicchi2013AbruptNetworks" class="csl-entry">
Radicchi, Filippo, and Alex Arenas. 2013. <span>“<span class="nocase">Abrupt transition in the structural formation of interconnected networks</span>.”</span> <em>Nature Physics</em> 9 (11): 717–20. <a href="https://doi.org/10.1038/nphys2761">https://doi.org/10.1038/nphys2761</a>.
</div>
<div id="ref-ramos2013" class="csl-entry">
Ramos, Erin M, Douglas Hoffman, Heather A Junkins, Donna Maglott, Lon Phan, Stephen T Sherry, Mike Feolo, and Lucia A Hindorff. 2013. <span>“Phenotype<span><span></span></span>Genotype Integrator (PheGenI): Synthesizing Genome-Wide Association Study (GWAS) Data with Existing Genomic Resources.”</span> <em>European Journal of Human Genetics</em> 22 (1): 144–47. <a href="https://doi.org/10.1038/ejhg.2013.96">https://doi.org/10.1038/ejhg.2013.96</a>.
</div>
<div id="ref-rehm2015" class="csl-entry">
Rehm, Heidi L., Jonathan S. Berg, Lisa D. Brooks, Carlos D. Bustamante, James P. Evans, Melissa J. Landrum, David H. Ledbetter, et al. 2015. <span>“ClinGen <span><span></span></span> The Clinical Genome Resource.”</span> <em>New England Journal of Medicine</em> 372 (23): 2235–42. <a href="https://doi.org/10.1056/nejmsr1406261">https://doi.org/10.1056/nejmsr1406261</a>.
</div>
<div id="ref-rual2005" class="csl-entry">
Rual, Jean-François, Kavitha Venkatesan, Tong Hao, Tomoko Hirozane-Kishikawa, Amélie Dricot, Ning Li, Gabriel F. Berriz, et al. 2005. <span>“Towards a Proteome-Scale Map of the Human Protein<span><span></span></span>protein Interaction Network.”</span> <em>Nature</em> 437 (7062): 1173–78. <a href="https://doi.org/10.1038/nature04209">https://doi.org/10.1038/nature04209</a>.
</div>
<div id="ref-Slack2013" class="csl-entry">
Slack, J. M. W. 2013. <span>“<span class="nocase">Molecular Biology of the Cell</span>.”</span> In <em>Principles of Tissue Engineering: Fourth Edition</em>. <a href="https://doi.org/10.1016/B978-0-12-398358-9.00007-0">https://doi.org/10.1016/B978-0-12-398358-9.00007-0</a>.
</div>
<div id="ref-Tanase2012MicroRNAs" class="csl-entry">
Tanase, Cristiana Pistol, Irina Ogrezeanu, and Corin Badiu. 2012. <span>“<span>MicroRNAs</span>.”</span> In <em>Molecular Pathology of Pituitary Adenomas</em>, 91–96. Elsevier. <a href="https://doi.org/10.1016/b978-0-12-415830-6.00008-1">https://doi.org/10.1016/b978-0-12-415830-6.00008-1</a>.
</div>
<div id="ref-uetz2000" class="csl-entry">
Uetz, Peter, Loic Giot, Gerard Cagney, Traci A. Mansfield, Richard S. Judson, James R. Knight, Daniel Lockshon, et al. 2000. <span>“A Comprehensive Analysis of Protein<span><span></span></span>protein Interactions in Saccharomyces Cerevisiae.”</span> <em>Nature</em> 403 (6770): 623–27. <a href="https://doi.org/10.1038/35001009">https://doi.org/10.1038/35001009</a>.
</div>
<div id="ref-Vazquez2003GrowingCorrelations" class="csl-entry">
Vázquez, Alexei. 2003. <span>“<span class="nocase">Growing network with local rules: Preferential attachment, clustering hierarchy, and degree correlations</span>.”</span> <em>Physical Review E</em> 67 (5): 056104. <a href="https://doi.org/10.1103/PhysRevE.67.056104">https://doi.org/10.1103/PhysRevE.67.056104</a>.
</div>
<div id="ref-wan2015" class="csl-entry">
Wan, Cuihong, Blake Borgeson, Sadhna Phanse, Fan Tu, Kevin Drew, Greg Clark, Xuejian Xiong, et al. 2015. <span>“Panorama of Ancient Metazoan Macromolecular Complexes.”</span> <em>Nature</em> 525 (7569): 339–44. <a href="https://doi.org/10.1038/nature14877">https://doi.org/10.1038/nature14877</a>.
</div>
<div id="ref-wishart2017" class="csl-entry">
Wishart, David S, Yannick D Feunang, An C Guo, Elvis J Lo, Ana Marcu, Jason R Grant, Tanvir Sajed, et al. 2017. <span>“DrugBank 5.0: A Major Update to the DrugBank Database for 2018.”</span> <em>Nucleic Acids Research</em> 46 (D1): D1074–82. <a href="https://doi.org/10.1093/nar/gkx1037">https://doi.org/10.1093/nar/gkx1037</a>.
</div>
<div id="ref-Wishart2006" class="csl-entry">
Wishart, David S, Craig Knox, An Chi Guo, Savita Shrivastava, Murtaza Hassanali, Paul Stothard, Zhan Chang, and Jennifer Woolsey. 2006. <span>“<span class="nocase">DrugBank: a comprehensive resource for in silico drug discovery and exploration.</span>”</span> <em>Nucleic Acids Research</em> 34 (Database issue): D668–72. <a href="https://doi.org/10.1093/nar/gkj067">https://doi.org/10.1093/nar/gkj067</a>.
</div>
<div id="ref-zhang2008" class="csl-entry">
Zhang, B., B.-H. Park, T. Karpinets, and N. F. Samatova. 2008. <span>“From Pull-down Data to Protein Interaction Networks and Complexes with Biological Relevance.”</span> <em>Bioinformatics</em> 24 (7): 979–86. <a href="https://doi.org/10.1093/bioinformatics/btn036">https://doi.org/10.1093/bioinformatics/btn036</a>.
</div>
<div id="ref-Zitnik2018" class="csl-entry">
Zitnik, Marinka, Monica Agrawal, and Jure Leskovec. 2018. <span>“<span class="nocase">Modeling polypharmacy side effects with graph convolutional networks</span>.”</span> In <em>Bioinformatics</em>. <a href="https://doi.org/10.1093/bioinformatics/bty294">https://doi.org/10.1093/bioinformatics/bty294</a>.
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="summary.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["NetMed.pdf", "NetMed.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
