<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Network Medicine</title>
  <meta name="description" content="Workshop in Advanced Bioinformatics: Network Medicine." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Network Medicine" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Workshop in Advanced Bioinformatics: Network Medicine." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Network Medicine" />
  
  <meta name="twitter:description" content="Workshop in Advanced Bioinformatics: Network Medicine." />
  

<meta name="author" content="Deisy Morselli Gysi" />


<meta name="date" content="2021-02-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="data-commonly-used-in-network-medicine.html"/>
<script src="libs/header-attrs-2.6.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Network Medicine</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="workshop-on-network-medicine.html"><a href="workshop-on-network-medicine.html"><i class="fa fa-check"></i><b>1</b> Workshop on Network Medicine</a>
<ul>
<li class="chapter" data-level="1.1" data-path="workshop-on-network-medicine.html"><a href="workshop-on-network-medicine.html#whatarenets"><i class="fa fa-check"></i><b>1.1</b> What are networks?</a></li>
<li class="chapter" data-level="1.2" data-path="workshop-on-network-medicine.html"><a href="workshop-on-network-medicine.html#terminology"><i class="fa fa-check"></i><b>1.2</b> Terminology</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="workshop-on-network-medicine.html"><a href="workshop-on-network-medicine.html#network-terminology"><i class="fa fa-check"></i><b>1.2.1</b> Network Terminology</a></li>
<li class="chapter" data-level="1.2.2" data-path="workshop-on-network-medicine.html"><a href="workshop-on-network-medicine.html#biological-terminology"><i class="fa fa-check"></i><b>1.2.2</b> Biological Terminology</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html"><i class="fa fa-check"></i><b>2</b> Data Commonly Used in Network Medicine</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#PPI"><i class="fa fa-check"></i><b>2.1</b> Protein Protein Interaction Networks</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#measuring-ppis"><i class="fa fa-check"></i><b>2.1.1</b> Measuring PPIs</a></li>
<li class="chapter" data-level="2.1.2" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#commonly-used-data-sources-for-ppis"><i class="fa fa-check"></i><b>2.1.2</b> Commonly used data sources for PPIs</a></li>
<li class="chapter" data-level="2.1.3" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#understanding-a-ppi"><i class="fa fa-check"></i><b>2.1.3</b> Understanding a PPI</a></li>
<li class="chapter" data-level="2.1.4" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#exercises"><i class="fa fa-check"></i><b>2.1.4</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#GDA"><i class="fa fa-check"></i><b>2.2</b> Gene Disease Association</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#commonly-used-data-sources-for-gdas"><i class="fa fa-check"></i><b>2.2.1</b> Commonly used data sources for GDAs</a></li>
<li class="chapter" data-level="2.2.2" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#understanding-a-gda-dataset"><i class="fa fa-check"></i><b>2.2.2</b> Understanding a GDA dataset</a></li>
<li class="chapter" data-level="2.2.3" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#exercises-1"><i class="fa fa-check"></i><b>2.2.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#DTs"><i class="fa fa-check"></i><b>2.3</b> Drug-Targets</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#properties-of-an-ideal-drug-target"><i class="fa fa-check"></i><b>2.3.1</b> Properties of an ideal drug target:</a></li>
<li class="chapter" data-level="2.3.2" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#commonly-used-data-sources-for-gdas-1"><i class="fa fa-check"></i><b>2.3.2</b> Commonly used data sources for GDAs</a></li>
<li class="chapter" data-level="2.3.3" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#understanding-a-drug-target-dataset"><i class="fa fa-check"></i><b>2.3.3</b> Understanding a Drug-Target dataset</a></li>
<li class="chapter" data-level="2.3.4" data-path="data-commonly-used-in-network-medicine.html"><a href="data-commonly-used-in-network-medicine.html#exercises-2"><i class="fa fa-check"></i><b>2.3.4</b> Exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>3</b> Methods fo Disease Module Identification and Disease Similarity</a>
<ul>
<li class="chapter" data-level="3.1" data-path="methods.html"><a href="methods.html#diseasemodule"><i class="fa fa-check"></i><b>3.1</b> Disease Module</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="methods.html"><a href="methods.html#example-in-real-data"><i class="fa fa-check"></i><b>3.1.1</b> Example in real data</a></li>
<li class="chapter" data-level="3.1.2" data-path="methods.html"><a href="methods.html#exercises-3"><i class="fa fa-check"></i><b>3.1.2</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="methods.html"><a href="methods.html#gene-overlap"><i class="fa fa-check"></i><b>3.2</b> Gene Overlap</a></li>
<li class="chapter" data-level="3.3" data-path="methods.html"><a href="methods.html#dissep"><i class="fa fa-check"></i><b>3.3</b> Disease Separation</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="methods.html"><a href="methods.html#example-in-real-data-1"><i class="fa fa-check"></i><b>3.3.1</b> Example in real data</a></li>
<li class="chapter" data-level="3.3.2" data-path="methods.html"><a href="methods.html#exercise"><i class="fa fa-check"></i><b>3.3.2</b> Exercise</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="method-for-drug-repurpusing.html"><a href="method-for-drug-repurpusing.html"><i class="fa fa-check"></i><b>4</b> Method for drug-repurpusing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="method-for-drug-repurpusing.html"><a href="method-for-drug-repurpusing.html#proximity"><i class="fa fa-check"></i><b>4.1</b> Proximity</a></li>
<li class="chapter" data-level="4.2" data-path="method-for-drug-repurpusing.html"><a href="method-for-drug-repurpusing.html#example-in-real-data-2"><i class="fa fa-check"></i><b>4.2</b> Example in real data</a></li>
<li class="chapter" data-level="4.3" data-path="method-for-drug-repurpusing.html"><a href="method-for-drug-repurpusing.html#exercises-4"><i class="fa fa-check"></i><b>4.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="summary.html"><a href="summary.html"><i class="fa fa-check"></i><b>5</b> Summary</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="./">Workshop in Advanced Bioinformatics</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Network Medicine</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Network Medicine</h1>
<p class="author"><em>Deisy Morselli Gysi</em></p>
<p class="date"><em>2021-02-03</em></p>
</div>
<div id="workshop-on-network-medicine" class="section level1" number="1">
<h1><span class="header-section-number">Chapter 1</span> Workshop on Network Medicine</h1>
<p>Network medicine (<strong>NetMed</strong>) is a field of Network Science that uses Systems Biology to understand its impact on medicine, and it is mostly focused on network topology. In NetMed we often use biological networks to represent the topological space and it be used to identify disease modules, their relationship to other diseases, drug repurpusing, and drug combinations. The biological networks often used for it are based on protein-protein interactions (PPI), Gene-Disease-Associations (GDA), and Drug-Targets.</p>
<p>In this workshop, we will learn:</p>
<ul>
<li>how to identify disease modules (Session <a href="methods.html#diseasemodule">3.1</a>);</li>
<li>how to predict disease comorbidities (Session <a href="methods.html#dissep">3.3</a>);</li>
<li>how to repurpuse drugs using a network approach (Session <a href="method-for-drug-repurpusing.html#proximity">4.1</a>).</li>
</ul>
<div id="whatarenets" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> What are networks?</h2>
<blockquote>
<p>Adapted from <span class="citation"><a href="references.html#ref-Gysi2020" role="doc-biblioref">Deisy Morselli Gysi and Nowick</a> (<a href="references.html#ref-Gysi2020" role="doc-biblioref">2020</a>)</span>.</p>
</blockquote>
<p>Network Science is broadly employed in many fields: from understanding <em>how friends bond in a party</em> to <em>how animals interact</em>; from <em>how superheroes appear in the same comic books</em> to <em>how genes can be related to a specific biological process</em>. Network analysis is especially beneficial for understanding complex systems, in all research fields. Examples of complex biological or medical systems include gene regulatory, ecological and neuropsychology networks. In this workshop, focus is given to applications of Network Science to the medical sciences.</p>
<p>Here, I will start by introducing the basic network terminologies and then explore how can we define and identify disease modules, identify disease commorbidities and lastly, we will learn how to repurpuse drugs for diseases with known modules. For each step, I will then present some classical and some new studies.</p>
<p>It is expected some degree of familiarity with <code>R</code>, <code>ggplot2</code>, <code>tidyr</code>, and <code>igraph</code>.</p>
</div>
<div id="terminology" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Terminology</h2>
<p>While the nature of each system, i.e. what its entities are and what kind of interactions they have, is different, there are common notations. A short review of common network terms can be found in Session <a href="workshop-on-network-medicine.html#network-terminology">1.2.1</a> and a brief review of biological terms can be found in Session <a href="workshop-on-network-medicine.html#biological-terminology">1.2.2</a>.</p>
<p>The set of interactions among a set of entities is, in general, called a graph or a network <span class="citation">(<a href="references.html#ref-Newman2018" role="doc-biblioref">Newman 2018</a>; <a href="references.html#ref-Barabasi2016" role="doc-biblioref">Albert-Láslzo Barabási 2016</a>)</span>. In graph theory, each entity is called a vertex, while in network notation it is called a node <span class="citation">(<a href="references.html#ref-Barabasi2016" role="doc-biblioref">Albert-Láslzo Barabási 2016</a>)</span>. Accordingly, the connections between two entities are called edges or links, respectively <span class="citation">(<a href="references.html#ref-Barabasi2016" role="doc-biblioref">Albert-Láslzo Barabási 2016</a>)</span>. In this workshop, I will always use the network notation, unless otherwise specified. The total number of nodes in a network is often denoted as <strong>N</strong> and the number of links in a network is denoted as <strong>L</strong>. While nodes can receive a label, links in general, are not labeled <span class="citation">(<a href="references.html#ref-Barabasi2016" role="doc-biblioref">Albert-Láslzo Barabási 2016</a>)</span> (although, in many cases, weights can also be perceived as a label). A network can be represented mathematically as an adjacency matrix (usually denoted as <strong>A</strong>) (Table <a href="workshop-on-network-medicine.html#tab:ADJ">1.1</a>), an edge-list (Table <a href="workshop-on-network-medicine.html#tab:EL">1.2</a>), or visually as a graph (Figure <a href="workshop-on-network-medicine.html#fig:GRAPH">1.1</a>).</p>
<table>
<caption><span id="tab:ADJ">Table 1.1: </span>Mathematical Representation of a Network: Adjacency Matrix</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">A</th>
<th align="right">C</th>
<th align="right">B</th>
<th align="right">D</th>
<th align="right">H</th>
<th align="right">F</th>
<th align="right">J</th>
<th align="right">I</th>
<th align="right">E</th>
<th align="right">G</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">B</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">H</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">F</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">J</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">I</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">E</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">G</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>Links of a network can possess a direction (normally depicted by an arrow), which indicates that the interaction is asymmetric, e.g. one gene is regulating another gene, or a person follows somebody else in a social network. Networks with directed links are called directed networks, while networks without directed interactions or in which the direction is not known are referred to as undirected networks, e.g. collaboration in the same study or interactions between proteins. In NetMed - and in this workshop - most of the times we assume that networks do not posses a direction. The links can also have a weight to express the strength of the interaction, which results in a weighted network <span class="citation">(<a href="references.html#ref-Newman2018" role="doc-biblioref">Newman 2018</a>; <a href="references.html#ref-Barabasi2016" role="doc-biblioref">Albert-Láslzo Barabási 2016</a>)</span>. Usually, the weight is graphically displayed as the thickness or the length of the links.</p>
<table>
<caption><span id="tab:EL">Table 1.2: </span>Mathematical Representation of a Network: Edge List</caption>
<thead>
<tr class="header">
<th align="left">Source</th>
<th align="left">Target</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="left">D</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="left">H</td>
</tr>
<tr class="odd">
<td align="left">B</td>
<td align="left">F</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="left">J</td>
</tr>
<tr class="odd">
<td align="left">A</td>
<td align="left">I</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="left">D</td>
</tr>
<tr class="odd">
<td align="left">A</td>
<td align="left">B</td>
</tr>
<tr class="even">
<td align="left">A</td>
<td align="left">E</td>
</tr>
<tr class="odd">
<td align="left">B</td>
<td align="left">G</td>
</tr>
<tr class="even">
<td align="left">A</td>
<td align="left">C</td>
</tr>
</tbody>
</table>
<p>Networks can also have different dimensions. These dimensions can be understood as layers (or different link types) of the same system <span class="citation">(<a href="references.html#ref-Kurant2006LayeredNetworks" role="doc-biblioref">Kurant and Thiran 2006</a>; <a href="references.html#ref-Kivela2014MultilayerNetworks" role="doc-biblioref">Kivela et al. 2014</a>)</span>. For example, in a multi-omics multilayer system, each layer can be constructed using different -omics data (for example genomics, transcriptomics, proteomics, etc.) where the ‘whole’ biological system can be understood as a network of networks <span class="citation">(<a href="references.html#ref-DeDomenico2017" role="doc-biblioref">De Domenico 2017</a>)</span>. The topology and the dynamic properties of the whole network can be changed by simply transforming the weights of the interactions, or by ignoring that nodes can interact in many ways <span class="citation">(<a href="references.html#ref-Mucha2010CommunityNetworks" role="doc-biblioref">Mucha et al. 2010</a>; <a href="references.html#ref-Radicchi2013AbruptNetworks" role="doc-biblioref">Radicchi and Arenas 2013</a>)</span> also ignoring the node’s importance to the system. We will not deal with multi-layer networks in this workshop.</p>
<div class="figure" style="text-align: center"><span id="fig:GRAPH"></span>
<img src="NetMed_files/figure-html/GRAPH-1.png" alt="Visual Representation of a Network: Graph" width="80%" />
<p class="caption">
Figure 1.1: Visual Representation of a Network: Graph
</p>
</div>
<div id="network-terminology" class="section level3" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Network Terminology</h3>
<ul>
<li><p>A <strong>network</strong> is a pair <strong>G = (N, L)</strong> of a set <strong>N</strong> of nodes connected by a set <strong>L</strong> of links.</p></li>
<li><p>Two nodes are <strong>neighbours</strong> if they are <strong>connected</strong>. The <strong>degree</strong> (d) of a node is the <strong>number of nodes</strong> it interacts with <span class="citation">(<a href="references.html#ref-Bondy2008GraphTheory" role="doc-biblioref">Bondy and Murty 2008</a>)</span>.</p></li>
<li><p>The <strong>weight</strong> is a measure of how strong a particular interaction is <span class="citation">(<a href="references.html#ref-Bondy2008GraphTheory" role="doc-biblioref">Bondy and Murty 2008</a>)</span>.</p></li>
<li><p>The <strong>strength</strong> of a node is the <strong>sum of the weights</strong> attached to links belonging to a node <span class="citation">(<a href="references.html#ref-Barrat2003TheNetworks" role="doc-biblioref">Barrat et al. 2003</a>)</span>.</p></li>
<li><p>The <strong>direction</strong> of a link specifies the source (starting point) and a target (endpoint) where the interaction occurs <span class="citation">(<a href="references.html#ref-Barabasi2016" role="doc-biblioref">Albert-Láslzo Barabási 2016</a>)</span> .</p></li>
<li><p><strong>Hubs</strong> are nodes with a <strong>much larger degree</strong> compared to the average degree value <span class="citation">(<a href="references.html#ref-Barrat2003TheNetworks" role="doc-biblioref">Barrat et al. 2003</a>)</span>.</p></li>
<li><p>A set of highly interconnected nodes is a <strong>module</strong> or <strong>cluster</strong> <span class="citation">(<a href="references.html#ref-Li2009" role="doc-biblioref">A. Li and Horvath 2009</a>)</span>. Two nodes are connected in a network, if a sequence of adjacent nodes, a <strong>path</strong>, connects them <span class="citation">(<a href="references.html#ref-barabasi2004network" role="doc-biblioref">Albert László Barabási and Oltvai 2004</a>)</span>.</p></li>
<li><p>The <strong>shortest path length</strong> is the number of links along the shortest path connecting two nodes <span class="citation">(<a href="references.html#ref-barabasi2004network" role="doc-biblioref">Albert László Barabási and Oltvai 2004</a>)</span>.</p></li>
<li><p>The <strong>average path length</strong> is the average of the shortest paths between all pairs of nodes <span class="citation">(<a href="references.html#ref-barabasi2004network" role="doc-biblioref">Albert László Barabási and Oltvai 2004</a>)</span>.</p></li>
<li><p>The <strong>diameter</strong> is the maximum distance between two nodes <span class="citation">(<a href="references.html#ref-Bondy2008GraphTheory" role="doc-biblioref">Bondy and Murty 2008</a>)</span>.</p></li>
<li><p>The <strong>modularity index</strong> is a measure of the strength of the network division into modules when this measure is maximized; it can be used for identifying nodes communities <span class="citation">(<a href="references.html#ref-Newman2018" role="doc-biblioref">Newman 2018</a>)</span>.</p></li>
<li><p><strong>Preferential attachment</strong> is the tendency of nodes to form new links preferentially to nodes with a high number of links <span class="citation">(<a href="references.html#ref-barabasi1999emergence" role="doc-biblioref">Albert László Barabási and Albert 1999</a>; <a href="references.html#ref-Vazquez2003GrowingCorrelations" role="doc-biblioref">Vázquez 2003</a>)</span>.</p></li>
<li><p>The probability that a random node in the network has a particular degree is given by the <strong>degree distribution</strong> <span class="citation">(<a href="references.html#ref-barabasi2004network" role="doc-biblioref">Albert László Barabási and Oltvai 2004</a>)</span>.</p></li>
<li><p>A <strong>bipartide graph</strong> is a network in which the nodes can be divided into two disjoint sets of nodes such that links connect nodes from the two sets to each other, but never inside the same set <span class="citation">(<a href="references.html#ref-Barabasi2016" role="doc-biblioref">Albert-Láslzo Barabási 2016</a>)</span>. In those networks, most of the network measures are calculated differently than in a unipartide network.</p></li>
<li><p>The <strong>clustering coefficient</strong> describes the degree with which a node is connected to all its neighbours <span class="citation">(<a href="references.html#ref-barabasi2004network" role="doc-biblioref">Albert László Barabási and Oltvai 2004</a>)</span>.</p></li>
<li><p>The <strong>global clustering</strong> coefficient measures the total number of triangles in a network <span class="citation">(<a href="references.html#ref-Barabasi2016" role="doc-biblioref">Albert-Láslzo Barabási 2016</a>)</span>.</p></li>
<li><p>The <strong>average clustering</strong> coefficient is the average of the clustering coefficient of all nodes in a network <span class="citation">(<a href="references.html#ref-barabasi2004network" role="doc-biblioref">Albert László Barabási and Oltvai 2004</a>)</span>.</p></li>
<li><p><strong>Centrality</strong> is a set of measures that have been proposed to help to define the most central nodes. It has many interpretations for autonomy, control, risk, exposure, influence and power <span class="citation">(<a href="references.html#ref-Borgatti2006ACentrality" role="doc-biblioref">Borgatti and Everett 2006</a>)</span>.</p>
<ul>
<li><p><strong>Closeness centrality</strong> is defined as the average distance from a single vertex to all other vertices<span class="citation">(<a href="references.html#ref-Newman2018" role="doc-biblioref">Newman 2018</a>)</span>.</p></li>
<li><p><strong>Betweenness centrality</strong> is defined as the total number of shortest paths between pairs of nodes that pass through a particular node <span class="citation">(<a href="references.html#ref-Newman2018" role="doc-biblioref">Newman 2018</a>)</span>.</p></li>
</ul></li>
<li><p><strong>Global measures</strong> are measures that describe the whole network, for example, <em>degree distribution; average clustering coefficient; path length; modularity index</em>.</p></li>
<li><p><strong>Local measures</strong> are characteristics of individual nodes of a network, such as their <em>degree</em> and <em>centrality</em>.</p></li>
</ul>
</div>
<div id="biological-terminology" class="section level3" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Biological Terminology</h3>
<ul>
<li><p><strong>DNA</strong> is the hereditary material of most organisms – usually all cells of an organism have the same DNA <span class="citation">(<a href="references.html#ref-Slack2013" role="doc-biblioref">Slack 2013</a>)</span>.</p></li>
<li><p><strong>Genes</strong> are the basic physical and functional units of heredity. They are parts of the DNA and contain the information for producing functional RNAs and proteins. <span class="citation">(<a href="references.html#ref-Slack2013" role="doc-biblioref">Slack 2013</a>)</span>.</p></li>
<li><p><strong>Proteins</strong> are large, complex molecules that play many critical roles in the body. The proteins are responsible for most of the work in cells and are necessary for structure, function, and regulation of the cells. They can act as enzymes, antibodies, transporters, transcription factors etc. <span class="citation">(<a href="references.html#ref-Slack2013" role="doc-biblioref">Slack 2013</a>)</span>.</p></li>
<li><p>The <strong>RNA</strong> is synthesized from the DNA but has different properties and functions than the DNA. Some RNAs carry out biological functions in a cell, while others, messenger RNA (mRNA), are turned into proteins that fulfil biological functions <span class="citation">(<a href="references.html#ref-Slack2013" role="doc-biblioref">Slack 2013</a>)</span>.</p></li>
<li><p>A <strong>non-coding RNA (ncRNA)</strong> is an RNA that does not encode a protein. NcRNAs often play a role in gene regulation <span class="citation">(<a href="references.html#ref-MattickNon-codingRNA" role="doc-biblioref">Mattick and Makunin 2006</a>)</span>.</p></li>
<li><p><strong>microRNAs (miRNA)</strong> are examples of ncRNA; they are involved in posttranscriptional regulation of protein expression <span class="citation">(<a href="references.html#ref-Tanase2012MicroRNAs" role="doc-biblioref">Tanase, Ogrezeanu, and Badiu 2012</a>)</span>.</p></li>
<li><p><strong>Gene expression</strong> is, in short, the coupled process of transcription (from DNA to RNA) and translation (from RNA to proteins) to transform the stored information inside the DNA into proteins <span class="citation">(<a href="references.html#ref-Slack2013" role="doc-biblioref">Slack 2013</a>)</span>.</p></li>
<li><p><strong>RNA-Seq</strong> is a technique used to sequence the RNAs in a sample. The result is the snapshot abundance of all RNAs expressed in the sample at a particular time, often called the transcriptome <span class="citation">(<a href="references.html#ref-Metzker2010SequencingGeneration" role="doc-biblioref">Metzker 2010</a>)</span>.</p></li>
<li><p><strong>Microarrays</strong>, or <strong>gene chips</strong>, are chips with thousands of tiny spots containing a known DNA sequence. It is used to measure the abundance of mRNAs by eminence of fluorescence <span class="citation">(<a href="references.html#ref-Slack2013" role="doc-biblioref">Slack 2013</a>)</span>.</p></li>
<li><p><strong>Transcription Factors</strong> are DNA binding proteins that activate or repress the transcription of particular target genes <span class="citation">(<a href="references.html#ref-Latchman1997TranscriptionOverview" role="doc-biblioref">Latchman 1997</a>)</span>.</p></li>
<li><p><strong>Gene Regulatory Factors</strong> are responsible for controlling the expression of genomic information and include transcription factors, co-factors, epigenetic modifiers, miRNAs and others <span class="citation">(<a href="references.html#ref-Hobert2008GeneMicroRNAs" role="doc-biblioref">Hobert 2008</a>)</span>.</p></li>
<li><p><strong>Systems Biology</strong> examines the structures and dynamics of cellular and organismal function, instead of isolated characteristics of a cell or organism.</p></li>
<li><p><strong>Drug repositioning</strong> (or drug repurposing) is the process of redeveloping a compound for use in a different disease.</p></li>
<li><p><strong>Yeast-Two-Hybrid (Y2H)</strong> systems is a system to measure protein-protein interaction. Two proteins to be tested for interaction are expressed in yeast; one protein is fused to a DNA-binding domain from a transcription factor while another protein (Y) is fused to a transcription activation domain. If X and Y interact, there will be a formation of a colony on media used as evidence of the interaction of X and Y <span class="citation">(<a href="references.html#ref-Parrish2006YeastMapping" role="doc-biblioref">Parrish, Gulyas, and Finley 2006</a>)</span>.</p></li>
<li><p><strong>Protein complex immunoprecipitation</strong> is an alternative method for measuring protein interactions. It involves immunoprecipitation of the protein bait, purification of the complex, and the identification of the interacting partners.</p></li>
<li><p><strong>High-throughput Mass Spectrometry</strong> has the ability to detect a characteristic mass to charge ratio of different substances in a sample. It is used to identify the proteins present in a sample <span class="citation">(<a href="references.html#ref-Kempa2019HighAnalysis" role="doc-biblioref">Kempa et al. 2019</a>)</span>.</p></li>
<li><p><strong>Chromatin immunoprecipitation followed by sequencing (ChIP–Seq)</strong> can be used to identify binding sites of transcription factors in the DNA or of histone modification in a genome-wide manner <span class="citation">(<a href="references.html#ref-Park2009ChIP-seq:Technology" role="doc-biblioref">Park 2009</a>)</span>.</p></li>
<li><p><strong>Chromatin Isolation by RNA Purification followed by sequencing (ChIRP-seq)</strong> maps lncRNA interactions to the chromatin <span class="citation">(<a href="references.html#ref-Park2009ChIP-seq:Technology" role="doc-biblioref">Park 2009</a>)</span>.</p></li>
<li><p><strong>Genome-wide association studies (GWAS)</strong> are studies where millions of SNPs are tested for association with a particular phenotype using hundreds or thousands of individuals. Those studies shed light on the genetic basis of complex traits.</p></li>
<li><p><strong>Omics</strong> is a term that refers to the study of different areas in biology, and indicates the totality of some kind, e.g. genome, transcriptome, proteome, etc.</p></li>
</ul>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="data-commonly-used-in-network-medicine.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["NetMed.pdf", "NetMed.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
